lib/api/manipulation.js

```diff
@@ -109,6 +110,40 @@ exports.prepend = _insert(function(dom, children, parent) {
   uniqueSplice(children, 0, 0, dom, parent);
 });
 
+exports.wrap = function(wrapper) {
+  var wrapperFn = typeof wrapper === 'function' && wrapper,
+      lastIdx = this.length - 1;
+
+  _.forEach(this, function(el, i) {
+    var parent = el.parent || el.root,
+        siblings = parent.children,
+        dom, index;
+
+    if (!parent) {
+      return;
+    }
+
+    if (wrapperFn) {
+      wrapper = wrapperFn.call(el, i);
+    }
+
+    if (typeof wrapper === 'string' && !isHtml(wrapper)) {
+      wrapper = this.parents().last().find(wrapper).clone();
+    }
+
+    dom = this._makeDomArray(wrapper, i < lastIdx).slice(0, 1);
+    index = siblings.indexOf(el);
+
+    updateDOM([el], dom[0]);
+    // The previous operation removed the current element from the `siblings`
+    // array, so the `dom` array can be inserted without removing any
+    // additional elements.
+    uniqueSplice(siblings, index, 0, dom, parent);
```

Comment:
I was game to try this, but one of the `wrap` tests specifies a function as the only argument and does not return a value. The simplification you suggested causes an error to be thrown in this case because it assumes `dom` is an array with length >= 1. The expected behavior here is undefined, but since jQuery doesn't throw (and since this is a code style change more than anything), I think it makes sense to leave it as-is.